src/Utilities/StdHelpers.hpp

```diff
@@ -283,3 +283,61 @@ std::string formatted_string(const std::string& fmt, Args... args) {
   }
   return rtn;
 }
+
+// Arithmetic operators for std::array<T, Dim>
+
+template <size_t Dim, typename T>
+inline void operator+=(std::array<T, Dim>& lhs, const std::array<T, Dim>& rhs) {
+  for (size_t i = 0; i < Dim; ++i) {
+    lhs[i] += rhs[i];
+  }
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator+(const std::array<T, Dim>& lhs,
+                                    const std::array<T, Dim>& rhs) {
+  std::array<T, Dim> result = lhs;
+  result += rhs;
+  return result;
+}
+
+template <size_t Dim, typename T>
+inline void operator-=(std::array<T, Dim>& lhs, const std::array<T, Dim>& rhs) {
+  for (size_t i = 0; i < Dim; ++i) {
+    lhs[i] -= rhs[i];
+  }
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator-(const std::array<T, Dim>& lhs,
+                                    const std::array<T, Dim>& rhs) {
+  std::array<T, Dim> result = lhs;
+  result-=rhs;
+  return result;
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator*(const std::array<T, Dim>& lhs,
+                                    const double scale) {
```

Comment:
Why `double`? I'd argue any type `U` such that `T{} * U{}` is defined is valid, for example I might want to multiply an `array` of `DataVector` by a `DataVector`